src/mps/mps.jl

```diff
@@ -585,9 +585,44 @@ psi = randomMPS(s, n->isodd(n) ? "Up" : "Dn"; linkdims=m)
 Cuu = correlation_matrix(psi,"Cdagup","Cup";site_range=2:8)
 ```
 """
-function correlation_matrix(psi::MPS, Op1::AbstractString, Op2::AbstractString; kwargs...)
+function correlation_matrix(psi::MPS, _Op1::AbstractString, _Op2::AbstractString; kwargs...)
   N = length(psi)
   ElT = promote_itensor_eltype(psi)
+  s = siteinds(psi)
+
+  Op1=_Op1 #make copies into which we can insert "F" string operators.
```

Comment:
Hi Jan, not sure these variables (`Op1` and `Op2` along with `_Op1` and `_Op2`) are needed. As you may know, this line just makes a reference to the same string, and then later a line like `Op1 = "$Op1*F"` just makes the variable `Op1` "point" to a new string elsewhere in memory while `_Op1` keeps pointing at the original string.

Reply:
Ok nevermind, I see now why these are needed, for the case of computing the lower triangle and fermionic operators. I was just thrown off by the comment which made it sound like it was a Julia string issue.